tests/ui_tools/test_boxes.py

```diff
@@ -743,88 +725,360 @@ def test_generic_autocomplete_user_mentions(
         )
 
     @pytest.mark.parametrize(
-        "text, state, required_typeahead, to_pin",
+        "text, state, required_typeahead, to_pin, to_mute",
```

Comment:
To avoid this adding lots of empty data and having lines spill-over, as well as improving the self-documenting of the test cases, I'm wondering if this would be clearer as a dict which might be:
```
{}  # no pins or mutes
{"pinned": [...]}  # equivalent to to_pin=[...]
{"muted": {...}}  # equivalent to to_mute={...}
```

If so, this logic could be a quick refactor commit before this with data and logic adjusted for just to_pin there, and then extended in this commit.